File: yugioh-20221012/src/main/java/tw/gaas/yugioh/domain/card/enu/Phase.java
package tw.gaas.yugioh.domain.card.enu;

public enum Phase {

    // 初始化
    INIT,

    // 抽牌
    LEFT_DRAW,
    // 召喚怪獸
    LEFT_MONSTER,
    // 發動魔法卡
    LEFT_SPELL,
    // 覆蓋陷阱卡
    LEFT_TRAP,
    // 攻擊
    LEFT_BATTLE,

    // 抽牌
    RIGHT_DRAW,
    // 召喚怪獸
    RIGHT_MONSTER,
    // 發動魔法卡
    RIGHT_SPELL,
    // 覆蓋陷阱卡
    RIGHT_TRAP,
    // 攻擊
    RIGHT_BATTLE,
}

File: yugioh-kehao/domain/src/main/kotlin/tw/gaas/yugioh/domain/enu/Phase.kt
package tw.gaas.yugioh.domain.enu

/**
 * 階段
 * - WAITING: 對局等待 (需要其他玩家加入)
 * - READY: 對局已準備
 * - DRAW_PHASE: 抽牌階段
 * - MAIN_PHASE_1: 主要階段 1
 * - BATTLE_PHASE: 戰鬥階段
 * - END_PHASE: 結束階段
 * - FINISHED: 對局結束
 */
enum class Phase(private val value: String) {
    WAITING("WAITING"),
    READY("READY"),
    DRAW_PHASE("DRAW_PHASE"),
    MAIN_PHASE_1("MAIN_PHASE_1"),
    BATTLE_PHASE("BATTLE_PHASE"),
    END_PHASE("END_PHASE"),
    FINISHED("FINISHED");
}
